### PR TITLE
updating 'PubSubExample' to latest api

### PR DESCRIPTION
**Pull request by jabubake**
PubSubExample is referenced in the READMEs, I updated the older version that got recently deleted in @pongad 's commit : https://github.com/GoogleCloudPlatform/google-cloud-java/commit/e27f4e4116528d17a294cf361e911b9bdf394b0e
to use the new APIs.

fyi, @SurferJeffAtGoogle

### CODE REVIEW

**Comment by lesv** [comment]
@jabubake Travis is unhappy.

**Comment by lesv** [comment]
@jabubake Travis is running 4 tests at the same time -- probably why it's unhappy.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java [hunk 1]
```diff
@@ -0,0 +1,1005 @@
...
+import com.google.cloud.Role;
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.pubsub.spi.v1.*;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.iam.v1.Binding;
+import com.google.iam.v1.Policy;
+import com.google.iam.v1.TestIamPermissionsResponse;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.*;
```
Comment (lesv) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java]: Is this used?

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java [hunk 2]
```diff
@@ -0,0 +1,1005 @@
...
+            System.out.printf("Expected: %s%n", action.params());
+            return;
+        } catch (Exception ex) {
+            System.out.println("Failed to parse arguments.");
+            ex.printStackTrace();
+            return;
+        }
+        action.run(arg);
+    }
+}
```
Comment (lesv) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java]: NewLine

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java (hunk 1 above)
Comment (jabubake) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java]: pubsub.v1.* -> covers the classes Subscription, Topic, TopicName, ProjectName, SubscriptionName etc.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java (hunk 2 above)
Comment (jabubake) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java]: added

**Pull request merged**